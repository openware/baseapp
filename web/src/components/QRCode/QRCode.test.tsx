import { render } from '@testing-library/react';
import React from 'react';
import { QRCode, QRCodeProps } from '.';

const renderComponent = (props: QRCodeProps) => render(<QRCode {...props} />);

describe('QRCode', () => {
    it('has qr-code className', () => {
        const { container } = renderComponent({ data: 'sevaerverv343', dimensions: 118 });
        expect(container.querySelector('.qr-code')).toBeInTheDocument();
    });

    it('renders correctly', () => {
        const { container } = renderComponent({ data: 'sevaerverv343', dimensions: 118 });
        expect(container).toMatchSnapshot();
    });
});
