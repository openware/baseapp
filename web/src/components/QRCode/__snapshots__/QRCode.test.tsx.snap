// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRCode renders correctly 1`] = `
<div>
  <div
    class="qr-code"
  >
    <svg
      height="118"
      viewBox="0 0 21 21"
      width="118"
    >
      <path
        d="M0,0 h21v21H0z"
        fill="#FFFFFF"
        shape-rendering="crispEdges"
      />
      <path
        d="M0 0h7v1H0zM9 0h3v1H9zM14,0 h7v1H14zM0 1h1v1H0zM6 1h1v1H6zM10 1h2v1H10zM14 1h1v1H14zM20,1 h1v1H20zM0 2h1v1H0zM2 2h3v1H2zM6 2h1v1H6zM8 2h4v1H8zM14 2h1v1H14zM16 2h3v1H16zM20,2 h1v1H20zM0 3h1v1H0zM2 3h3v1H2zM6 3h1v1H6zM8 3h1v1H8zM10 3h3v1H10zM14 3h1v1H14zM16 3h3v1H16zM20,3 h1v1H20zM0 4h1v1H0zM2 4h3v1H2zM6 4h1v1H6zM8 4h1v1H8zM10 4h1v1H10zM12 4h1v1H12zM14 4h1v1H14zM16 4h3v1H16zM20,4 h1v1H20zM0 5h1v1H0zM6 5h1v1H6zM8 5h2v1H8zM11 5h1v1H11zM14 5h1v1H14zM20,5 h1v1H20zM0 6h7v1H0zM8 6h1v1H8zM10 6h1v1H10zM12 6h1v1H12zM14,6 h7v1H14zM8 7h1v1H8zM10 7h1v1H10zM0 8h1v1H0zM2 8h5v1H2zM9 8h3v1H9zM14 8h5v1H14zM0 9h3v1H0zM8 9h3v1H8zM12 9h2v1H12zM15 9h4v1H15zM20,9 h1v1H20zM1 10h1v1H1zM3 10h2v1H3zM6 10h1v1H6zM8 10h2v1H8zM12 10h2v1H12zM17 10h3v1H17zM1 11h3v1H1zM7 11h3v1H7zM11 11h1v1H11zM13 11h1v1H13zM15,11 h6v1H15zM0 12h1v1H0zM2 12h1v1H2zM4 12h3v1H4zM8 12h1v1H8zM12 12h3v1H12zM17 12h1v1H17zM8 13h4v1H8zM16,13 h5v1H16zM0 14h7v1H0zM9 14h2v1H9zM12 14h1v1H12zM14 14h1v1H14zM19 14h1v1H19zM0 15h1v1H0zM6 15h1v1H6zM8 15h3v1H8zM12 15h2v1H12zM15 15h5v1H15zM0 16h1v1H0zM2 16h3v1H2zM6 16h1v1H6zM8 16h1v1H8zM10 16h6v1H10zM19,16 h2v1H19zM0 17h1v1H0zM2 17h3v1H2zM6 17h1v1H6zM8 17h1v1H8zM10 17h2v1H10zM13 17h1v1H13zM16 17h1v1H16zM18 17h1v1H18zM0 18h1v1H0zM2 18h3v1H2zM6 18h1v1H6zM8 18h2v1H8zM12 18h1v1H12zM14 18h1v1H14zM17 18h2v1H17zM0 19h1v1H0zM6 19h1v1H6zM9 19h3v1H9zM17 19h2v1H17zM0 20h7v1H0zM8 20h1v1H8zM12 20h3v1H12zM19 20h1v1H19z"
        fill="#000000"
        shape-rendering="crispEdges"
      />
    </svg>
  </div>
</div>
`;
